docs/advanced/api/index.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Advanced API
+---
+
+# Getting Started
+
+::: warning
+This guide lists advanced APIs to run tests via a Node.js script. If you just want to [run tests](/guide/), you probably don't need this. It is primarily used by library authors.
+:::
+
+## startVitest
+
+```ts
+function startVitest(
+  mode: VitestRunMode,
+  cliFilters: string[] = [],
+  options: CliOptions = {},
+  viteOverrides?: ViteUserConfig,
+  vitestOptions?: VitestOptions,
+): Promise<Vitest>
+```
+
+You can start running Vitest tests using its Node API:
+
+```js
+import { startVitest } from 'vitest/node'
+
+const vitest = await startVitest('test')
+
+await vitest.close()
+```
+
+`startVitest` function returns [`Vitest`](/advanced/api/vitest) instance if tests can be started.
+
+If watch mode is not enabled, Vitest will call `close` method automatically.
+
+If watch mode is enabled and the terminal supports TTY, Vitest will register console shortcuts.
+
+You can pass down a list of filters as a second argument. Vitest will run only tests that contain at least one of the passed-down strings in their file path.
+
+Additionally, you can use the third argument to pass in CLI arguments, which will override any test config options. Alternatively, you can pass in the complete Vite config as the fourth argument, which will take precedence over any other user-defined options.
+
+After running the tests, you can get the results from the [`state.getTestModules`](/advanced/api/test-module) API:
+
+```ts
+import type { TestModule } from 'vitest/node'
+
+const vitest = await startVitest('test')
+
+console.log(vitest.state.getTestModules()) // [TestModule]
+```
+
+::: tip
+The ["Running Tests"](/advanced/guide/tests#startvitest) guide has a usage example.
+:::
+
+## createVitest
+
+```ts
+function createVitest(
+  mode: VitestRunMode,
+  options: UserConfig,
+  viteOverrides: ViteUserConfig = {},
+  vitestOptions: VitestOptions = {},
+): Promise<Vitest>
+```
+
+You can create Vitest instance by using `createVitest` function. It returns the same [`Vitest`](/advanced/api/vitest) instance as `startVitest`, but it doesn't start tests and doesn't validate installed packages.
+
+```js
+import { createVitest } from 'vitest/node'
+
+const vitest = await createVitest('test', {
+  watch: false,
+})
+```
+
+::: tip
+The ["Running Tests"](/advanced/guide/tests#createvitest) guide has a usage example.
+:::
+
+## resolveConfig
+
+```ts
+function resolveConfig(
+  options: UserConfig = {},
+  viteOverrides: ViteUserConfig = {},
+): Promise<{
+  vitestConfig: ResolvedConfig
+  viteConfig: ResolvedViteConfig
+}>
+```
```

Comment:
This one is not showing the entrypoint its exposed from. What do you think about adding entrypoints as filenames here? 

<img src="https://github.com/user-attachments/assets/5c677fb4-9013-412b-b75a-e1d14ce624b3" width="400" />

Reply:
I don't think the file name is noticible here. All other APIs just have an example. This whole page is for exports from `vitest/node` - we should add a sentence about it at the start of the article